### PR TITLE
 Add github action for unit tests 

### PR DESCRIPTION
**Pull request by exyi**
When build or unit tests fail, it's quite annying to go though the Azure CI,
so I have added a simple github action that will say that.

It also checks that we can build DotVVM.Framework without warnings

### CODE REVIEW

File: src/Framework/Framework/Hosting/ErrorPages/ErrorPageTemplate.cs [hunk 1]
```diff
@@ -37,12 +37,16 @@ public class ErrorPageTemplate : IErrorWriter
         public string Summary { get; }
         public IErrorSectionFormatter[] Formatters { get; }
 
-        public void WriteText(string str)
+        public void WriteText(string? str)
```
Comment (quigamdev) [inline src/Framework/Framework/Hosting/ErrorPages/ErrorPageTemplate.cs:40]: This seems like an unrelated change. Why these changes are here?

Comment (exyi) [reply src/Framework/Framework/Hosting/ErrorPages/ErrorPageTemplate.cs:40]: This only fixes the warnings that we now check in CI

**Review by quigamdev — APPROVED** [review]
